lib/connection.js

```diff
@@ -341,14 +341,16 @@ Connection.prototype.ready = function(n){
  * @api private
  */
 
-Connection.prototype.finish = function(id){
+Connection.prototype.finish = function(id, fn){
   assertValidMessageId(id);
-  var fn = this.onerror;
   var self = this;
-  if (!this._ready) return fn(new Error('cannot finish, connection not ready'));
+  if (!this._ready) return this.onerror(new Error('cannot finish, connection not ready'));
   this.command('FIN', [id], function(){
```

Comment:
Can `FIN` error? (The param may be omitted here just because it's unused.) I doubt it but it's worth double checking